frontend/src/app/core/services/team.service.ts

```diff
@@ -4,33 +4,74 @@ import { INewTeam } from '@core/models/INewTeam';
 import { ITeam } from '@core/models/ITeam';
 import { IUpdateTeam } from '@core/models/IUpdateTeam';
 import { HttpInternalService } from '@core/services/http-internal.service';
-import { Observable } from 'rxjs';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
 
 @Injectable({
     providedIn: 'root',
 })
 export class TeamService {
     public routePrefix = '/team';
 
-    // eslint-disable-next-line no-empty-function
-    constructor(private httpService: HttpInternalService) {}
+    private emitTeamCreationSource = new Subject<ITeam>();
+
+    public teamCreationEmitted$ = this.emitTeamCreationSource.asObservable();
+
+    private emitTeamDeletionSource = new Subject<number>();
+
+    public teamDeletionEmitted$ = this.emitTeamDeletionSource.asObservable();
+
+    private currentTeamSource = new BehaviorSubject<number | undefined>(undefined);
+
+    public currentTeamEmitted$ = this.currentTeamSource.asObservable();
+
+    constructor(private httpService: HttpInternalService) {
+        this.getCurrentUserTeams()
+            .subscribe(teams => {
+                if (teams.length > 0) {
```

Comment:
Why in the constructor?

Reply:
Because ngOninit cannot be used for injectable service

Reply:
it is strange to load data in service

Reply:
I need it to set first current team. Header cannot be used to set the first current team because it relaunches after page changing. I believe we can put all subscriptions and this method to something like team-subscription service. What do you think about it?

Reply:
No, we can do some data sharing between component by service
But we need to reload teams in the header component

Reply:
Okay, reloading in header is already implemented with subscriptions

Reply:
Can I leave it in constructor?

Reply:
Fixed